packages/tpetra/core/src/Tpetra_CrsGraph_decl.hpp

```diff
@@ -1346,6 +1366,18 @@ namespace Tpetra {
     /// allocate memory.
     virtual void
     removeEmptyProcessesInPlace (const Teuchos::RCP<const map_type>& newMap);
+
+    /// \brief Get information about row and column map matching
+    ///
+    /// \warning This method is only for expert users.
+    /// \warning We make no promises about backwards compatibility
+    ///   for this method. It may disappear or change at any time.
+    ///
+    /// If the check is true, the global ids for the local row index i would
+    /// and local column index i would be the same.
+    virtual bool
+    doRowAndColumnMapsMatch ( ) const;
```

Comment:
I realize now that Tpetra already has a predicate for this, called `isLocallyFitted`.  It currently lives in the `Tpetra::Details` namespace.  I'm not sure what you would prefer to call this predicate, but it's definitely a local (per-MPI-process) property, so the method name should have the word "local" in it.

Reply:
In fact, the predicate even has a unit test:

https://github.com/trilinos/Trilinos/blob/f0350316239aaf41e8e6b82378612aaedf9cc72c/packages/tpetra/core/test/Map/Map_isLocallyFitted.cpp#L107

Reply:
Oh, that's nice! Plus, it's a lot smarter than the suggested code as it checks for the contiguity and can make decisions using that. That actually makes it cheap in many situations.

How about I then simply make it use `LocalMap` and add add Kokkos lambdas to it? This way, we don't even need to touch the graph. However, I would prefer to move it out of the `Details` namespace. I could also further optimize it as LocalMap has first contiguous index and last, and we can skip part of the check using those.